pkg/golinters/gochecknoglobals.go

```diff
@@ -1,100 +1,30 @@
 package golinters
 
 import (
-	"fmt"
-	"go/ast"
-	"go/token"
-	"strings"
-	"sync"
+	"flag"
 
 	"golang.org/x/tools/go/analysis"
 
+	"4d63.com/gochecknoglobals/checknoglobals"
+
 	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
-	"github.com/golangci/golangci-lint/pkg/lint/linter"
-	"github.com/golangci/golangci-lint/pkg/result"
 )
 
-const gochecknoglobalsName = "gochecknoglobals"
-
-//nolint:dupl
 func NewGochecknoglobals() *goanalysis.Linter {
-	var mu sync.Mutex
-	var resIssues []goanalysis.Issue
-
-	analyzer := &analysis.Analyzer{
-		Name: gochecknoglobalsName,
-		Doc:  goanalysis.TheOnlyanalyzerDoc,
-		Run: func(pass *analysis.Pass) (interface{}, error) {
-			var res []goanalysis.Issue
-			for _, file := range pass.Files {
-				fileIssues := checkFileForGlobals(file, pass.Fset)
-				for i := range fileIssues {
-					res = append(res, goanalysis.NewIssue(&fileIssues[i], pass))
-				}
-			}
-			if len(res) == 0 {
-				return nil, nil
-			}
+	// gochecknoglobals only lints test files if the `-t` flag is passed so we
+	// set up our own FlagSet and add it to the analyzer before running it. This
+	// can be turned of by using the regular golangci-lint flags such as
+	// `--tests` or `--skip-files`.
+	flags := flag.NewFlagSet("", flag.ContinueOnError)
+	flags.Bool("t", true, "Include tests")
 
-			mu.Lock()
-			resIssues = append(resIssues, res...)
-			mu.Unlock()
+	gochecknoglobals := checknoglobals.Analyzer()
+	gochecknoglobals.Flags = *flags
 
-			return nil, nil
-		},
-	}
 	return goanalysis.NewLinter(
-		gochecknoglobalsName,
-		"Checks that no globals are present in Go code",
-		[]*analysis.Analyzer{analyzer},
+		gochecknoglobals.Name,
+		gochecknoglobals.Doc,
+		[]*analysis.Analyzer{gochecknoglobals},
 		nil,
```

Comment:
```suggestion
		map[string]map[string]interface{}{gochecknoglobals.Name: {"t": true}},
```

Reply:
This is the way to do it! It's bee na while since I looked at this code so I missed this simple solution. Will submit patch removing the flag overwriting in favour of this later today.